### PR TITLE
Pre-release. version v1.7.0

### PR DESCRIPTION
**Pull request by Joeper214**
Features
-------

- Select arguments in Serverspec generator #108 
- selectable ec2  #119 
- Optional detach EC2 #120 

Bug fix
-------

- Escape error message #116 
- Add EC2 Validator #113 

Version Up
--------

- Vue 1.0.7    #108 
- Bundle update on 2015-11-18  #123  

Others
--------

- Add EC2 validation #113 

Update process
----------

```sh
cd YOUR_SKYHOPPER_DIRECTORY
export RAILS_ENV=production  # This command isn't needed if you use develop environment.
git pull
git checkout <SkyHopper Version>
bundle install
bower install
cd frontend/
# npm i
gulp ts
cd ..
```

### CODE REVIEW

File: app/controllers/resources_controller.rb [hunk 1]
```diff
@@ -49,6 +49,16 @@ def create
...
     end
 
+    unless infra.instance(physical_id).describe_keypair == infra.keypairname
+      # TODO: I18n
+      render text: "#{physical_id}: Keypair does not matched to this Infrastructure!", status: 400; return
+    end
+
+    unless infra.instance(physical_id).status != :terminated
+      # TODO: I18n
+      render text: "Cannot add #{physical_id}. if status is terminated.", status: 400; return
```
Comment (skyarch-networks) [inline app/controllers/resources_controller.rb]: __[brakeman]__
* Unescaped parameter value rendered inline [:ref](http://brakemanscanner.org/docs/warning_types/cross_site_scripting/)

File: app/controllers/resources_controller.rb [hunk 2]
```diff
@@ -49,6 +49,16 @@ def create
       render text: "Cannot find #{physical_id}", status: 400; return
     end
 
+    unless infra.instance(physical_id).describe_keypair == infra.keypairname
+      # TODO: I18n
+      render text: "#{physical_id}: Keypair does not matched to this Infrastructure!", status: 400; return
```
Comment (skyarch-networks) [inline app/controllers/resources_controller.rb]: __[brakeman]__
* Unescaped parameter value rendered inline [:ref](http://brakemanscanner.org/docs/warning_types/cross_site_scripting/)

**Comment by Joeper214** [comment]
Can't update this branch.

**Comment by Joeper214** [comment]
maybe because of the fix that recently merged to master? @pocke

**Comment by pocke** [comment]
#111 is merged into master, I'll fix this problem

**Comment by Joeper214** [comment]
Please Merge @pocke  after #119 is merged.

**Comment by pocke** [comment]
:ok_hand:

**Pull request merged**